app/util/variables.ts

```diff
@@ -204,4 +204,33 @@ export function getVariablesForCollection(
       versionId: collection.version_id, variables, coordinateVariables });
   }
   return variableInfo;
+}
+
+/**
+ * Validate that the extend query parameter is both a valid variable name or concept ID
+ * and falls within the requested variable(s) (OGC collectionId) and collection(s).
+ * Assumes that the extend variable only needs to be found within one of the requested collections.
+ * @param extendParam - The name or concept ID of the variable to be extended (extend query parameter)
+ * @param collectionIdParam - The OGC collectionId
+ * @param eosdisCollections - An array of CMR collections
+ * @param varInfos - The variables requested by the user
+ * @throws RequestValidationError if the variable cannot be extended for the user's request
+ */
+export function validateExtend(extendParam: string, collectionIdParam: string, eosdisCollections: CmrCollection[], varInfos: VariableInfo[]): void {
+  if (collectionIdParam === 'all') {
+    for (const collection of eosdisCollections) { // Look through all of the variables
+      const foundVar = collection.variables.some((v) => doesPathMatch(v, extendParam));
```

Comment:
It might be worth specifically limiting the variables checked to those that are dimensions of others. The successful example in the ticket is a bit confusing - `green_var` isn't a dimension variable, it is a 3-D science variable, so I'm not sure it makes sense to be able to request to extend it.

Some possible options might be:

* Limiting to the `coordinateVariables`. Potential issues include: non 1-D coordinate variables (e.g., 2-D lon or lat variables in swath data). Whether all extendable dimensions have `VariableType = "COORDINATE"`.
* Check the length of the `Dimensions` attribute of the UMM-Var record. If it has 1 element, then it is 1-D, and dimension variables are 1-D. Downsides: some collections have 1-D science variables (e.g., trajectory data), and it relies on metadata curators populating the `dimensions` attribute for UMM-Var records.
* When all the variables are pulled from CMR, iterate through all of them and create a set of all the `Name` values of the elements of `Dimensions` for each variable. Downside: Again relies on well populated UMM-Var records.

I think the three options get more complicated as you go down the list, but I'm tempted to say that the last option is most likely to get the right set of variables that are dimensions.

Reply:
Thanks for the feedback Owen. I'm going to get this merged ASAP after making the change to allow possibly multiple values for `extend`. We'll implement the validation checks after things have solidified a bit. That way we're not doing anything prematurely but we also won't be blocking anything on the service side of things.